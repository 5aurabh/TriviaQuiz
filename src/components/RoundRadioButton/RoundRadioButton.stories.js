import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { linkTo } from '@storybook/addon-links';
import RoundRadioButton from './index.js';
import 'font-awesome/css/font-awesome.min.css';

storiesOf('RoundRadioButton', module)
  .add('default', () => <RoundRadioButton />)
  .add('with custom label', () => <RoundRadioButton label="Milk" />)
  .add('with default checked state', () => (
    <RoundRadioButton label="I am already checked" checked={true} />
  ))
  .add('with custom change handler', () => (
    <RoundRadioButton label="Click me" onChange={action('change handler')} />
  ));
