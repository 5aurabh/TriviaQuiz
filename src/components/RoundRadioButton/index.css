.RoundRadioButtonContainer {
  display: inline-block;
  margin: 0 5px;
}

.RoundRadioButton {
  opacity: 0;
  position: absolute;
}

.RoundRadioButtonLabel {
  display: inline-block;
  vertical-align: middle;
  margin: 5px;
  cursor: pointer;
  font-family: sans-serif;
}

.RoundRadioButton + .RoundRadioButtonLabel:before {
  content: '';
  background: #fff;
  border: 2px solid #ddd;
  display: inline-block;
  vertical-align: middle;
  width: 20px;
  height: 20px;
  padding: 2px;
  margin-right: 10px;
  text-align: center;
}

.RoundRadioButton + .RoundRadioButtonLabel:before {
  border-radius: 50%;
}

.RoundRadioButton:checked + .RoundRadioButtonLabel:before {
  content: '\f00c';
  font-family: 'FontAwesome';
  color: #fff;
  background-color: #15e9c0;
  border: 2px solid #15e9c0;
}
