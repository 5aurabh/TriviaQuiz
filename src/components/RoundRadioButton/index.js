// @flow
import React from 'react';
import styles from './index.css';
type Props = {
  checked: boolean,
  name: ?string,
  id: ?string,
  label: ?string,
  value: ?string,
  onChange: () => *,
};
const RoundRadioButton = (props: Props) => (
  <div className={styles.RoundRadioButtonContainer}>
    <input
      type="radio"
      id={props.id}
      name={props.name}
      value={props.value}
      defaultChecked={props.checked}
      className={styles.RoundRadioButton}
      onChange={props.onChange}
    />
    <label className={styles.RoundRadioButtonLabel} for={props.id}>
      {props.label}
    </label>
  </div>
);
/* Add default props */
RoundRadioButton.defaultProps = {
  checked: false,
  id: 'id-0',
  name: 'name-round-0',
  label: 'Yes',
  value: 'some-val',
  onChange: () => {},
};

export default RoundRadioButton;
