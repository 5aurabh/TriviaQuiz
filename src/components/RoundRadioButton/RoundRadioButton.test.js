import React from 'react';
import renderer from 'react-test-renderer';
import RoundRadioButton from './index.js';

it('RoundRadioButton: default', () => {
  const component = renderer.create(<RoundRadioButton />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
