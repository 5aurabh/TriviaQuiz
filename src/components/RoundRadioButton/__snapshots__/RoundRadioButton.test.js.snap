// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundRadioButton: default 1`] = `
<div
  className={undefined}
>
  <input
    className={undefined}
    defaultChecked={false}
    id="id-0"
    name="name-round-0"
    onChange={[Function]}
    type="radio"
    value="some-val"
  />
  <label
    className={undefined}
    htmlFor="id-0"
  >
    Yes
  </label>
</div>
`;
